### ISSUE TITLE
TMDB scraper still not working in 16.7.04

### ISSUE THREAD

**Issue by lelin88** [koying/SPMC#1193]
@koying  installed 16.7.04 and TMDB scraper still not working
Cannot connect to remote server ....

**Comment by lelin88** [comment]
@koying    ok, now working. After installation 16.7.04 we must update the TMDB scraper in addons
if no update, no working.After installation it doesn't update itself

**Comment by koying** [comment]
Should auto-update at some point

**Comment by djneiljbirmingham** [comment]
How do I update on Fire TV, as there is no apk yet?  Any help appreciated thanks.

**Comment by pawhtiobo** [comment]
For those who use the Windows version (16.5.5), I found a fix, just delete the TMDB folders in:

C:\Program Files (x86)\SPMC\addons
C:\Users\(your username)\AppData\Roaming\Kodi\addons
C:\Users\(your username)\AppData\Roaming\Kodi\cache\scrapers

Reinstall the TMDB plugin from official Kodi repo.

**Comment by ThomMark** [comment]
@ djneiljbirmingham
We just need to wait a day or two for the official release, i have tested it on a none Android device and it is working. 
It will be soon here pal 😉

**Comment by djneiljbirmingham** [comment]
Thanks so much. 👍🏻

**Comment by grummel2005** [comment]
working here after update of the addons
thx a lot
perfect

**Comment by ninjageezer** [comment]
On my nvidia shield the SPMC on it i downloaded from the playstore is 16..7.03 will there be a fix for this or will there be an update on the playstore,or do we need to download one of the beta test versions ?

**Comment by Anda006** [comment]
The latest playstore version is 16.7.4. Update from there (you may need to restart the box if the update doesn't show in the Play Store) then update the TMDB plugin from the new repo (from within SPMC) and it'll work.

**Comment by rwhome** [comment]
it works for movies
but i still have problems with series(tv) 
no addon update available for tv from semperax repository

spmc 16.7.4 on firetv

**Comment by ThomMark** [comment]
@ Rwhome
Which version are you on mate?
Mine is working perfect and not needed changing since November 2017.

**Comment by ninjageezer** [comment]
ok cool maybe mine did not auto update i checked my version from the playstore and it was16.7.3

**Comment by rwhome** [comment]
@ThomMark 
i made the update after it stopped working today (or yesterday)
it was spmc version 16.7.2 before
now it is the actual spmc 16.7.4 version 
and version 1.3.10 for the tmdb tv series addon (from the kodi repository) which ist not working
and version 674.1.3 for the tmdb movie addon (from the semperpax repository) which is working
i think we will need also a special version for the tv series addon and not only the movie addon

i can switch to tvdb as scraper but i don't want to rebuild my whole tv series database :(

**Comment by ThomMark** [comment]
For those using .APK applications, ie: Firestick, the current working versions are:
TVDB: Version 672.0.5 (**2.05**)
TMDB: Version 674.1.3 (**4.1.2**)
You **must** also update to SPMC Version **16.7.04**

If either are lower than that version, you **must** update them.
Hope this helps 1 or 2 people who are still experiencing problems.

Mark

**Comment by rwhome** [comment]
@ThomMark 
this tvdb version you mentioned is actual not available in the semperpax repository on the firetv box/stick

p.s.
**sorry
i mixed it up with tmdb
tvdb is the version you mentioned**

**Comment by ThomMark** [comment]
@ rwhome 
I'm just going to grab some tea, then I will play with it and let you know how I can get the version to you.
I may have to zip it and send it over, then you can install via ZIP file.

I'll get back to you in half hour mate.

**Comment by ThomMark** [comment]
Actually, did you try updating via Settings, My Addon's, All, then TVDB and update manually?
If so, fine, I will get a work around for you. If not, can you try mate?

Mark

**Comment by rwhome** [comment]
@ThomMark 

sorry , a missunderstanding
you speak from tvdb
i am not using tvdb
i want to use tmdb as scraper like before for the tv series

**Comment by ThomMark** [comment]
Yes, and TMDB: Version **674.1.3 (4.1.2)** works on **SPMC 16.7.4**
You MUST update SPMC first, then manually update the scraper mate.

**Comment by rwhome** [comment]
it works well for movies
but it not works for tv series
this are two different tmdb addons we speak about

**Comment by ThomMark** [comment]
I understand.
Are you saying your chosen MOVIE scrapper is TMDB?
But the version is NOT 674.1.3 (4.1.2)?

You will not have to re-build your items, I just need to understand exactly what is adrift so I can help you fix it.

**Comment by rwhome** [comment]
no
for the movie scraper i use tmdb version 674.1.3 (semperpax repository) and it is working
for the tv series scraper i use tmdb version 1.3.10 (kodi repository) which is not working
this are two different scraper addons for two different types of media
the tmdb tv series scraper addon from the kodi repository has stopped working
and there is no actualized tmdb version for the tmdb tv series scraper addon from koying
it is possible to use the tvdb scraper addon instead
but this is not what i want
because in this case i have to make a rebuild for my series database

**Comment by ThomMark** [comment]
Got ya! 
You could insert your own TMDB API Key.
That would be easy enough, after all, it's the MPAA that's blocked the last key.
Hence all this trouble

**Comment by rwhome** [comment]
i maybe do that if nothing happens in the next days
maybe @koying will also update the tmdb addon for tv series
anyway ... thanks for the support

**Comment by ThomMark** [comment]
I will mess with mine here for you and get it working, I'll update you

UPDATE: Hi Mate
I could not find a working API key in the public domain nor via updating.
I did copy my API key from TMDB into my TMDB/TV and it worked fine, it updated everything.
I then reversed the key back, and was unable to connect to network.

So if you want a quick fix, it's easy enough to do, and techinically it won't get blocked as only you have the key.
Let me know if you need any guidance how to copy the key.

Mark

**Comment by Beefy314** [comment]
Thank you it's working now on Shield and Fire TV 👍

**Comment by proteus32** [comment]
Working since update on Shield. Thanks a lot great job, save my week:)

**Comment by ninjageezer** [comment]
ok so i have restarted shield checked playstore and the latest version on offer for me is only 16.07.03  not the 04 version ?? what am i doing wrong please ?

**Comment by ninjageezer** [comment]
all sorted yeahhh playstore updated SPMC to 16.07.04

**Comment by idunkle** [comment]
I have my own API key for TMDB, but where, within SPMC (and I guess Kodi) do I add it ?
I cant seem to see any settings as such under TMDB addon

**Comment by ThomMark** [comment]
Do you use a FTP client to connect to your SPMC?
If so, I can talk you through editing the DATA script in the system, but before I launch into how to do it, its best checking that you can FTP to the **com.semperpax.spmc16** folder first.

**Comment by idunkle** [comment]
I dont use an FTP client but could do if needed and able to copy things to and from Shield and edit files etc that are on it so should be able to edit the script if knew what to do

Can certainly edit and view files in the folder

and thanks for offer of help, very kind

**Comment by idunkle** [comment]
Can certainly access the folder, just checked

**Comment by rwhome** [comment]
the api-keys should be in the file(s) tmdb.xml (firetv in my case)
/Android/data/com.semperpax.spmc16/files/.spmc/addons/metadata.themoviedb.org
/Android/data/com.semperpax.spmc16/files/.spmc/addons/metadata.tvshows.themoviedb.org

@ThomMark 
i haven't changed the api-key
i made a reset for my tv show database entries
changed scraper to 'local information only' and then back to 'the movie database' scraper
maybe a coincidence
but tmdb for tv shows stopped working at the same time like tmdb for movies
but the key from kodi seems to be ok ... it is working now
i don't know .... very confusing

if this kind of troubles continue it is maybe better to use own keys

is this the right place to generate key(s) ?
https://developers.themoviedb.org/3/getting-started/introduction
